pkg/resources/management.cattle.io/v3/project/validator.go

```diff
@@ -143,6 +151,42 @@ func (a *admitter) admitCommonCreateUpdate(oldProject, newProject *v3.Project) (
 	return admission.ResponseAllowed(), nil
 }
 
+func (a *admitter) validateContainerDefaultResourceLimit(limit *v3.ContainerResourceLimit) error {
+	if limit == nil {
+		return nil
+	}
+	fieldPath := projectSpecFieldPath.Child(containerLimitField)
+	requestsCPU, err := parseResource(limit.RequestsCPU)
+	if err != nil {
+		return field.Invalid(fieldPath, limit.RequestsCPU, fmt.Sprintf("failed to parse container default requested CPU: %s", err))
+	}
+	limitsCPU, err := parseResource(limit.LimitsCPU)
+	if err != nil {
+		return field.Invalid(fieldPath, limit.LimitsCPU, fmt.Sprintf("failed to parse container default CPU limit: %s", err))
+	}
+	requestsMemory, err := parseResource(limit.RequestsMemory)
+	if err != nil {
+		return field.Invalid(fieldPath, limit.RequestsMemory, fmt.Sprintf("failed to parse container default requested memory: %s", err))
+	}
+	limitsMemory, err := parseResource(limit.LimitsMemory)
+	if err != nil {
+		return field.Invalid(fieldPath, limit.LimitsMemory, fmt.Sprintf("failed to parse container default memory limit: %s", err))
+	}
+	if resources := negativeResources(requestsCPU, limitsCPU, requestsMemory, limitsMemory); len(resources) > 0 {
```

Comment:
One alternative way of doing this would be to add a side function that adds a value to a list if not nil, and do the check here instead:
```suggestion
         resourceList := make(v1.ResourceList)
        addIfNotNil(resourceList, v1.ResourceRequestsCPU, requestsCPU)
        addIfNotNil(resourceList, v1.ResourceLimitsCPU, limitsCPU)
        addIfNotNil(resourceList, v1.ResourceRequestsMemory, requestsMemory)
        addIfNotNil(resourceList, v1.ResourceLimitsMemory, limitsMemory)
	if resources := quota.IsNegative(resourceList); len(resources) > 0 {
```